### PR TITLE
Vs/ast

### PR DESCRIPTION
**Pull request by vlad-scherbich**
This is the error that I get:

```
VSCHERBICH-2294:src vscherbich$ ocamlc -c oscar.ml
File "oscar.ml", line 5, characters 20-21:
Error: This expression has type int but an expression was expected of type
         string
```

Any ideas what's wrong here? I've compared this to the `calculator` code from HW1, and everything looks right. Thanks for any tips!

### CODE REVIEW

File: src/oscar.ml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            | Bit_RShift -> v1 lsr v2
+            | Bit_LShift -> v1 lsl v2
+        )
+    | Unop(op, e) ->
+        let v = eval e in
+        (
+          match op with
+              Neg -> -v
+            | Not -> !v
+            (* todo: i couldn't find Bit_Not in OCaml *)
```
Comment (vlad-scherbich) [inline src/oscar.ml]: Bit_Not is not listed in the Bitwise operations section here https://caml.inria.fr/pub/docs/manual-ocaml/libref/Pervasives.html

Comment (jibben) [reply src/oscar.ml]: So maybe we just don't have bit_not, that's not a big deal

**Comment by Howon** [comment]
Is it done?

File: src/oscar.ml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        )
+    (* todo: make it right *)
+    (* 
+        | Assign(x, e) -> symbol_table[x] = (eval_string e)
+        | Call(f, el) -> smth smth
+    *)
+
+let _ =
+    let lexbuf = Lexing.from_channel stdin in
+    let expr = Parser.expr Scanner.token lexbuf in
```
Comment (vlad-scherbich) [inline src/oscar.ml:99]: Well, it works up to this point:

```
ocamlc -c oscar.ml
<warning, warning, warning>
File "oscar.ml", line 99, characters 15-26:
Error: Unbound value Parser.expr
```

Comment (Howon) [reply src/oscar.ml:99]: is the type error from before fixed?

File: src/oscar.ml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+          match op with
+            Not -> not v
+        )
+    (* todo: make it right *)
+    (* 
+        | Assign(x, e) -> symbol_table[x] = (eval_string e)
+        | Call(f, el) -> smth smth
+    *)
+
+let rec eval_int = function
```
Comment (Howon) [inline src/oscar.ml]: What about floating point operations?

File: src/oscar.ml (hunk 2 above)
Comment (vlad-scherbich) [reply src/oscar.ml:99]: yes, the type error is fixed

Comment (Howon) [reply src/oscar.ml:99]: this is not supposed to be there anyway

**Pull request closed without merging**